src/main/java/org/vietabroader/view/MainView.java

```diff
@@ -408,25 +428,43 @@ public void update(Observable o, Object arg) {
         switch (currentStatus) {
             case SIGNED_OUT:
                 resetOnSignedOut();
+                setEnabledChildren(false);
+                panSignIn.setEnabled(true);
                 break;
             case SIGNED_IN:
                 btnAuthenticate.setText(BUTTON_TEXT_SIGN_OUT);
                 lblAuthMessage.setText(currentState.getUserEmail());
+                setEnabledChildren(false);
+                panSignIn.setEnabled(true);
+                panSpreadsheet.setEnabled(true);
                 break;
             case CONNECTED:
                 VASpreadsheet currentSpreadsheet = currentState.getSpreadsheet();
                 String spreadsheetTitle = currentSpreadsheet.getSpreadsheetTitle();
                 lblSpreadsheetMessage.setBackground(Color.GREEN);
                 lblSpreadsheetMessage.setForeground(Color.BLACK);
                 lblSpreadsheetMessage.setText("Connected to: " + spreadsheetTitle);
-
+                setEnabledChildren(true);
                 List<String> sheets = currentSpreadsheet.getSheetTitles();
                 cbbSheet.removeAllItems();
                 sheets.forEach(cbbSheet::addItem);
                 break;
             case QR_READING:
+                setEnabledChildren(false);
+                break;
+            case REFRESHED:
+                setEnabledChildren(true);
+                panWebcam.setEnabled(true);
                 break;
         }
     }
+
+    public void setEnabledChildren(boolean b) {
+        panSignIn.setEnabled(b);
+        panSpreadsheet.setEnabled(b);
+        panWorkspace.setEnabled(b);
+        panWebcam.setEnabled(b);
```

Comment:
Let's comment this line out for now. We're still working on the camera controller so it's pretty inconvenient to connect to a sheet every time we need to test the webcam.